File: sdk/php/src/Schemas/UnknownSchema.php
<?php

declare(strict_types=1);

namespace AnyVali\Schemas;

use AnyVali\ParseResult;
use AnyVali\Schema;
use AnyVali\ValidationContext;

final class UnknownSchema extends Schema
{
    public function getKind(): string
    {
        return 'unknown';
    }

    protected function validateValue(mixed $value, ValidationContext $ctx): ParseResult
    {
        return ParseResult::ok($value);
    }

    public function exportNode(): array
    {
        return ['kind' => 'unknown'];
    }
}

File: sdk/csharp/src/AnyVali/Schemas/UnknownSchema.cs
namespace AnyVali.Schemas;

public sealed class UnknownSchema : Schema<object?>
{
    internal override object? Validate(object? input, ValidationContext ctx) => input;

    internal override Dictionary<string, object?> ToNode()
    {
        var node = new Dictionary<string, object?> { ["kind"] = "unknown" };
        AddDefaultAndCoercion(node);
        return node;
    }

    internal override Schema Clone() => new UnknownSchema
    {
        DefaultValue = DefaultValue, CoercionCfg = CoercionCfg, IsPortable = IsPortable,
    };
}
